digits/scheduler.py

```diff
@@ -214,6 +214,34 @@ def get_job(self, job_id):
             return None
         return self.jobs.get(job_id, None)
 
+    def get_related_jobs(self, job):
+        """
+        Look through self.jobs to try to find the Jobs
+        whose parent contains job
+        """
+        related_jobs = []
+
+        if isinstance(job, ModelJob):
+            datajob = job.dataset
+        elif isinstance(job, DatasetJob):
+            datajob = job
+        else:
+            raise ValueError("Unhandled job type %s" % job.job_type())
+
+        for j in self.jobs:
```

Comment:
you can iterate over dictionary values:
```
for j in self.jobs.values():
```
That would be slightly more pythonic and you can replace `self.jobs[j]` with just `j` in the lines below